src/Components/AddField/index.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useContext } from 'react';
+import { addNewReception } from '../../services/api';
+import { DoctorContext } from '../../contexts/DoctorContext';
+import { validName, valiDate } from '../../validators';
+import './index.css';
+
+const AddField = ({ setAddedData, setError }) => {
+  const [newReception, setNewReception] = useState([{
+    patientName: '', doctorId: '', date: '', complaint: '',
+  }]);
+  const setInputedData = (inputData) => {
+    setNewReception({ ...newReception, ...inputData });
+  };
+
+  const setNewValues = async () => {
+    try {
+      const errors = [];
+      const body = newReception;
+      const {
+        patientName, date, complaint, doctorId,
+      } = body;
+      if (!patientName.trim() || !date || !complaint.trim() || !doctorId) {
+        return setError('All fields are required');
+      }
+      if (!validName(patientName)) errors.push('Enter valid patient name.');
+      if (!valiDate(date)) errors.push('Invalid date');
+      if (!complaint.trim()) errors.push('Enter valid text format');
+      if (errors.length) {
+        return setError(errors);
+      }
+      const data = await addNewReception(body);
+      setAddedData(data);
+    } catch (error) {
+      console.error(error.message);
```

Comment:
don't console error, show it somewhere. you can change it's message. don't make user to wait for response if there's error